docs/root/intro/arch_overview/arch_overview.rst

```diff
@@ -7,6 +7,7 @@ Architecture overview
   intro/intro
   listeners/listeners_toc
   http/http
+  matching/matching
```

Comment:
nit: I might consider moving this inside advanced? or at least lower in the list?